command.go

```diff
@@ -338,21 +338,34 @@ func (cmd *Command) setupSubcommand() {
 	cmd.flagCategories = newFlagCategoriesFromFlags(cmd.allFlags())
 }
 
+func (cmd *Command) hideHelp() bool {
+	tracef("hide help (cmd=%[1]q)", cmd.Name)
+	hideHelp := cmd.HideHelp
+
+	for parent := cmd.parent; parent != nil; parent = parent.parent {
+		hideHelp = hideHelp || parent.HideHelp
+	}
+
+	return hideHelp
```

Comment:
Simplify the code and improve performance.

```suggestion
	for c := cmd; c != nil; c = c.parent {
		if c.HideHelp {
			return true
		}
	}

	return false
```